### PR TITLE
Update filters to work with custom Ransackers

### LINKED ISSUE #3510: default filter should list custom ransackers
If there is a custom runsacker defined, like this:

``` ruby
class User < ActiveRecord::Base
  ransacker :fullname do
    Arel.sql("CONCAT(`firstname`, ' ', `lastname`)")
  end
end
```

There should be a automatic generated filter for `fullname`, like the filters for each other field.

### PR DESCRIPTION
**Pull request by whatcould**
It seems that ransackers (eg `ransacker :name_reversed`) are _not_ expected to be called without predicates (that is, they **are** called with predicates, eg: `name_reversed_cont`). So a ransacker filter without a predicate should not `seems_searchable?`.

Seems to me that custom ransackers would never have worked as intended with this code — an artifact of converting from metasearch without testing it in a sample application.

[fixes #3510 / #3409]

### CODE REVIEW

**Comment by Fivell** [comment]
@whatcould , can you rebase upstream ?

**Comment by whatcould** [comment]
@Fivell rebased!

**Comment by Fivell** [comment]
@whatcould , please rebase again and it will be merged.

File: spec/unit/filters/filter_form_builder_spec.rb [hunk 1]
```diff
@@ -424,7 +424,7 @@ def view.a_helper_method
     it "should work as string" do
       body = Capybara.string(filter :custom_title_searcher, as: :string)
       expect(body).to have_selector("input[name='q[custom_title_searcher_contains]']")
-      expect(body).to have_selector("input[name='q[custom_searcher_starts_with]']")
+      expect(body).to have_selector("input[name='q[custom_title_searcher_starts_with]']")
```
Comment (Fivell) [inline spec/unit/filters/filter_form_builder_spec.rb]: bad merge... that should be selector for ```option``` , not ```input``` :disappointed:

File: spec/unit/filters/filter_form_builder_spec.rb [hunk 2]
```diff
@@ -423,8 +423,8 @@ def view.a_helper_method
 
     it "should work as string" do
       body = Capybara.string(filter :custom_title_searcher, as: :string)
-      expect(body).to have_selector("input[name='q[custom_title_searcher_contains]']")
-      expect(body).to have_selector("input[name='q[custom_title_searcher_starts_with]']")
+      expect(body).to have_selector("option[name='q[custom_title_searcher_contains]']")
+      expect(body).to have_selector("option[name='q[custom_title_searcher_starts_with]']")
```
Comment (Fivell) [inline spec/unit/filters/filter_form_builder_spec.rb]: finally this should work

```
  it "should work as string" do
      body = Capybara.string(filter :custom_title_searcher, as: :string)
      expect(body).to have_selector("option[value=custom_title_searcher_contains]")
      expect(body).to have_selector("option[value=custom_title_searcher_starts_with]")
    end
```

**Pull request merged**